core/templates/dev/head/pages/exploration_editor/simple_editor_tab/SimpleEditorSidebarDirective.js

```diff
@@ -21,12 +21,34 @@ oppia.directive('simpleEditorSidebar', [function() {
     restrict: 'E',
     templateUrl: 'simpleEditor/sidebar',
     controller: [
-        '$scope', 'EditorModeService', 'SimpleEditorManagerService',
-        function($scope, EditorModeService, SimpleEditorManagerService) {
+       '$scope', 'EditorModeService', 'SimpleEditorManagerService',
+        'ScrollSyncService', 'QuestionIdService',
+        function($scope, EditorModeService, SimpleEditorManagerService,
+                 ScrollSyncService, QuestionIdService) {
       $scope.SUBFIELD_LABELS = [
         'Multiple choice', 'Correct answer', 'Hints', 'Bridge text'];
       $scope.setEditorModeToFull = EditorModeService.setModeToFull;
       $scope.questionList = SimpleEditorManagerService.getQuestionList();
+      $scope.ID_PREFIX = QuestionIdService.SIDEBAR_PREFIX;
+      $scope.getSidebarItemId = function(question, subfieldLabel) {
+        return QuestionIdService.getSidebarItemId(
+          question.getStateName(), subfieldLabel
+        );
+      };
+      $scope.scrollToField = function(question, subfieldLabel) {
+        ScrollSyncService.scrollTo(
+          QuestionIdService.getSubfieldId(
+            question.getStateName(), subfieldLabel
+          )
+        );
+      };
+      $scope.scrollToHeader = ScrollSyncService.scrollTo;
+      $scope.scrollToQuestion = function(question) {
+        ScrollSyncService.scrollTo(question.getId());
+      };
+      $scope.$on('SimpleEditorSidebarToggleCollapse', function() {
+        $scope.$apply();
```

Comment:
I am not sure i understood this one, what the $broadcast is for is to update the simple-editor-sidebar, because even if the value of question.collapsed has changed, the sidebar doesn't pick it up and does not update the view.